bundle/direct/dresources/model_serving_endpoint.go

```diff
@@ -0,0 +1,311 @@
+	}
+	_, err := r.client.ServingEndpoints.PutAiGateway(ctx, req)
+	if err != nil {
+		return fmt.Errorf("failed to update AI gateway: %w", err)
+	}
+	return nil
+}
+
+// TODO(shreyas): Add acceptance tests for update and for unsetting. Will be done once we add selective updates for these fields.
+func (r *ResourceModelServingEndpoint) updateConfig(ctx context.Context, id string, config *serving.EndpointCoreConfigInput) error {
+	if config == nil {
+		// Unset config in resource.
+		req := serving.EndpointCoreConfigInput{
+			Name:              id,
+			AutoCaptureConfig: nil,
+			ServedEntities:    nil,
+			TrafficConfig:     nil,
+			ServedModels:      nil,
+		}
+		_, err := r.client.ServingEndpoints.UpdateConfig(ctx, req)
+		return err
+	}
+	req := serving.EndpointCoreConfigInput{
+		Name:              id,
+		AutoCaptureConfig: config.AutoCaptureConfig,
+		ServedEntities:    config.ServedEntities,
+		TrafficConfig:     config.TrafficConfig,
+		ServedModels:      config.ServedModels,
+	}
+	_, err := r.client.ServingEndpoints.UpdateConfig(ctx, req)
+	if err != nil {
+		return fmt.Errorf("failed to update config: %w", err)
+	}
+	return nil
+}
+
+func (r *ResourceModelServingEndpoint) updateNotifications(ctx context.Context, id string, notifications *serving.EmailNotifications) error {
+	req := serving.UpdateInferenceEndpointNotifications{
+		Name:               id,
+		EmailNotifications: notifications,
+	}
+	_, err := r.client.ServingEndpoints.UpdateNotifications(ctx, req)
+	if err != nil {
+		return fmt.Errorf("failed to update notifications: %w", err)
+	}
+	return nil
+}
+
+func diffTags(currentTags, desiredTags []serving.EndpointTag) (addTags []serving.EndpointTag, deleteTags []string) {
+	addTags = make([]serving.EndpointTag, 0)
+
+	// build maps for easy lookup.
+	currentTagsMap := make(map[string]string)
+	desiredTagsMap := make(map[string]string)
+	for _, tag := range currentTags {
+		currentTagsMap[tag.Key] = tag.Value
+	}
+	for _, tag := range desiredTags {
+		desiredTagsMap[tag.Key] = tag.Value
+	}
+
+	// Compute keys to be added.
+	for key, desiredValue := range desiredTagsMap {
+		v, ok := currentTagsMap[key]
+		if !ok {
+			addTags = append(addTags, serving.EndpointTag{
+				Key:             key,
+				Value:           desiredValue,
+				ForceSendFields: nil,
+			})
+			continue
+		}
+		if v != desiredValue {
+			addTags = append(addTags, serving.EndpointTag{
+				Key:             key,
+				Value:           desiredValue,
+				ForceSendFields: nil,
+			})
+		}
+	}
+
+	// Compute keys to be deleted.
+	for key := range currentTagsMap {
+		if _, ok := desiredTagsMap[key]; !ok {
+			deleteTags = append(deleteTags, key)
+		}
+	}
+
+	return addTags, deleteTags
+}
+
+func (r *ResourceModelServingEndpoint) updateTags(ctx context.Context, id string, tags []serving.EndpointTag) error {
+	endpoint, err := r.client.ServingEndpoints.GetByName(ctx, id)
+	if err != nil {
+		return err
+	}
+
+	addTags, deleteTags := diffTags(endpoint.Tags, tags)
```

Comment:
Why are we diffing tags here?

What happens if remove old tag from databricks.yml? (do we have a test for that?)